### PR TITLE
Feature/add kb entry for weak mac algo

### PR DESCRIPTION
**Pull request by nmasdoufi-ol**
Add KB entry for Weak MAC Algorithm.

### CODE REVIEW

**Review by ybadaoui-ostorlab — APPROVED** [review]

File: WEB_SERVICE/WEB/_HIGH/WEAK_MAC_ALGORITHM/recommendation.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
+ssl_prefer_server_ciphers off;
+```
+
+3. **IIS:**
+```powershell
+New-Item 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.2\Server' -Force
+New-ItemProperty -path 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.2\Server' -name 'Enabled' -value '1' -PropertyType 'DWord' -Force
+New-ItemProperty -path 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.2\Server' -name 'DisabledByDefault' -value 0 -PropertyType 'DWord' -Force
+```
```
Comment (elyousfi5) [inline WEB_SERVICE/WEB/_HIGH/WEAK_MAC_ALGORITHM/recommendation.md:43]: i think its better to put like these: 

image


and please test in in a ticket comment to see if the rendering is correct.

Comment (nmasdoufi-ol) [reply WEB_SERVICE/WEB/_HIGH/WEAK_MAC_ALGORITHM/recommendation.md]: Done. Thanks.

File: WEB_SERVICE/WEB/_HIGH/WEAK_MAC_ALGORITHM/description.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+- CBC-MAC: The Cipher Block Chaining MAC can be vulnerable if not used with proper padding or if the underlying block cipher is weak. 
+- CMAC with weak block ciphers: While CMAC (Cipher-based Message Authentication Code) is secure when using strong block ciphers like AES, using it with weak ciphers (like DES) can lead to vulnerabilities. 
+- Poly1305 with weak key lengths: While Poly1305 is generally secure, using weak or small key sizes can compromise its security.
+- GMAC (Galois Message Authentication Code): While generally secure when used with Galois/Counter Mode (GCM), misuse or incorrect implementation can lead to vulnerabilities. 
+- KMAC: Although based on SHA-3, its use with insufficient key lengths can render it insecure.
+
+These algorithms have various weaknesses that can be exploited by attackers, potentially leading to:
+
+1. Collision attacks
+2. Preimage attacks
```
Comment (3asm) [inline WEB_SERVICE/WEB/_HIGH/WEAK_MAC_ALGORITHM/description.md:20]: What is this?

File: WEB_SERVICE/WEB/_HIGH/WEAK_MAC_ALGORITHM/description.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+3. Length extension attacks
```
Comment (3asm) [inline WEB_SERVICE/WEB/_HIGH/WEAK_MAC_ALGORITHM/description.md:21]: Would be nice to have a short explanation next to each.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: WEB_SERVICE/WEB/_HIGH/WEAK_MAC_ALGORITHM/recommendation.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+1. **Disable Weak MAC Algorithms:**
+   - Disable support for weak MAC algorithms such as MD5 and SHA1.
+   - Enable only strong MAC algorithms like SHA256, SHA384, or SHA512.
+
+2. **Configure Strong Cipher Suites:**
+   - Use cipher suites that incorporate strong MAC algorithms.
+   - Disable cipher suites that use weak MAC algorithms.
+
+3. **Implement Secure TLS Configuration:**
+   - Follow industry best practices for TLS configuration, such as those provided by Mozilla's SSL Configuration Generator or OWASP's TLS Cheat Sheet.
```
Comment (adnaneserrar) [inline WEB_SERVICE/WEB/_HIGH/WEAK_MAC_ALGORITHM/recommendation.md:12]: i would add a link for the cheat sheet

**Review by 3asm — APPROVED** [review]

**Pull request merged**